lib/sqlalchemy/util/queue.py

```diff
@@ -63,31 +72,25 @@ def __init__(self, maxsize=0, use_lifo=False):
         # If this queue uses LIFO or FIFO
         self.use_lifo = use_lifo
 
+    @atomic
     def qsize(self):
         """Return the approximate size of the queue (not reliable!)."""
 
-        self.mutex.acquire()
-        n = self._qsize()
-        self.mutex.release()
-        return n
+        return self._qsize()
 
+    @atomic
```

Comment:
these should look like Python 3's queue.py, will fix

Reply:
It looks like python3's queue. `atomic` is the decorator define above.

Reply:
source is from here: https://github.com/python/cpython/blob/master/Lib/queue.py#L91 .

Reply:
I think atomic more clear and safety because not needed to think about scope. If you see atomic then this method thread-safe.

Reply:
thanks, IMO this is too much verbiage for a small part of a single file.   if there were a concurrency framework in use throughout all of SQLAlchemy, and there was no use of "with mutex:" and instead it were all `@atomic` decorators, that would be more consistent.  however, it's inconsistent that one part of the library do a certain task in an entirely different style than the rest of the library.  Additionally these methods are ported directly from the Python 3 source code and I'm not even sure they are being used.